src/core/CompactionQueue.java

```diff
@@ -539,7 +544,10 @@ private KeyValue buildCompactedColumn(ByteBufferList compacted_qual,
       }
 
       final KeyValue first = row.get(0);
-      return new KeyValue(first.key(), first.family(), cq, cv);
+      if(tsdb.getConfig().getBoolean("tsd.storage.use_otsdb_timestamp"))
```

Comment:
This should be cached somewhere to avoid the duplicate calls to lookup and deserialize the value.